src/integrationTest/sam.test.ts

```diff
@@ -0,0 +1,192 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { mkdirpSync, readFileSync, removeSync } from 'fs-extra'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { SamLambdaRuntime } from '../../src/lambda/models/samLambdaRuntime'
+import { getSamCliContext } from '../../src/shared/sam/cli/samCliContext'
+import { runSamCliInit, SamCliInitArgs } from '../../src/shared/sam/cli/samCliInit'
+import { activateExtension, sleep, TIMEOUT } from './integrationTestsUtilities'
+
+const projectFolder = `${__dirname}`
+let projectSDK = ''
+let projectPath = ''
+let debuggerType = ''
+let documentUri: vscode.Uri
+
+const runtimes = [
+    { name: 'nodejs8.10', path: 'testProject/hello-world/app.js', debuggerType: 'node2' },
+    { name: 'nodejs10.x', path: 'testProject/hello-world/app.js', debuggerType: 'node2' },
+    { name: 'python2.7', path: 'testProject/hello_world/app.py', debuggerType: 'python' },
+    { name: 'python3.6', path: 'testProject/hello_world/app.py', debuggerType: 'python' },
+    { name: 'python3.7', path: 'testProject/hello_world/app.py', debuggerType: 'python' },
+    { name: 'dotnetcore2.1', path: 'testProject/src/HelloWorld/Function.cs', debuggerType: 'coreclr' }
+]
+
+async function openSamProject(): Promise<vscode.Uri> {
+    const documentPath = path.join(projectFolder, projectPath)
+    await vscode.workspace.openTextDocument(documentPath)
+
+    return vscode.Uri.file(documentPath)
+}
+
+function tryRemoveProjectFolder() {
+    try {
+        removeSync(path.join(projectFolder, 'testProject'))
+    } catch (e) {}
+}
+
+async function getCodeLenses(): Promise<vscode.CodeLens[]> {
+    let codeLenses: vscode.CodeLens[] | undefined
+    while (true) {
+        try {
+            // this works without a sleep locally, but not on CodeBuild
+            await sleep(200)
+            const codeLensesPromise: Thenable<vscode.CodeLens[] | undefined> = vscode.commands.executeCommand(
```

Comment:
Is the `codeLensesPromise` const necessary? Seems like you could directly await vscode.commands.executeCommand

Reply:
I'm not sure why it was like that, fixed